react-components/toolbar-filter/step2.md

```diff
@@ -1,61 +1,141 @@
-} from &quot;@patternfly/react-core/dist/esm/experimental&quot;;
-import Input from &quot;./components/input&quot;;
-import CheckboxSelect from &quot;./components/checkboxSelect&quot;;
-import Kebab from &quot;./components/kebab&quot;;
-import AppTable from &quot;./components/table&quot;;
-
-class ToolbarDemoApp extends React.Component {
-  constructor(props) {
-    super(props);
-  }
-
-  render() {
-    return (
-      &lt;React.Fragment&gt;
-        Toolbar Demo
-      &lt;/React.Fragment&gt;
-    );
-  }
+The individual components placed inside of a toolbar are each wrapped in a `DataToolbarItem` component. `DataToolbarItems` can be passed an optional `variant` prop to format a specific type of component it wraps such as a search filter, a pagination component, a bulk select, or an overflow menu. 
+
+Wrapping components in the `DataToolbarItem` component and adding the proper variant ensures the components are spaced appropriately.
+
+Use the state of the `ToolbarDemoApp` to manage the three applied filters in the `DataToolbar`.
+
+1) **Locate the following constructor in `ToolbarDemoApp**
+
+<pre class="file">
+constructor(props) {
+  super(props);
+}
+</pre>
+
+2) **Initialize the state in the constructor located in step 1**
+
+Add a `filters` object to the state. `filters` should have three properties (name, risk, and status) each with an empty array as its value. This filters object will manage the three filters in the `DataToolbar`.
+
+Once step 2 is completed, the constructor should look like the snippet below.
+
+<pre class="file">
+constructor(props) {
+  super(props);
+  this.state = {
+    filters: {
+      name: [],
+      risk: [],
+      status: []
+    }
+  };
+}
+</pre>
+
+3) **Add two event handlers to the `ToolbarDemoApp` class**
+
+Both of the event handlers can be inserted outside and below the constructor modified in the previous steps.
+
+The first will be an `onSearch` event handler which will fire when the user interacts with the name filter.
+
+<pre class="file" data-target="clipboard">
+onSearch = (value) =&gt; {
+  this.setState(prevState =&gt; {
+    return {
+      filters: {
+        ...prevState.filters,
+        name: [...prevState.filters.name, value]
+      }
+    };
+  });
+}
+</pre>
+
+The second event handler will be an `onSelect` event handler which will fire when the user interacts with the risk and status filters.
+
+<pre class="file" data-target="clipboard">
+onSelect = (type, event, selection) =&gt; {
+  const checked = event.target.checked;
+  this.setState(prevState =&gt; {
+    const prevSelections = prevState.filters[type.toLowerCase()];
+    return {
+      filters: {
+        ...prevState.filters,
+        [type.toLowerCase()]: checked
+          ? [...prevSelections, selection]
+          : prevSelections.filter(value => value !== selection)
+      }
+    };
+  });
 };
+</pre>
+
+4) **Add the following constants inside the top of the render function above the return statement**
```

Comment:
This has a lot of instructions before introducing what the purpose of this step is. Change to something like:
`Declare constants in the render function.`

And then specify the location